### PR TITLE
feat: knowledge tab list

### PR DESCRIPTION
**Pull request by gozineb**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

## Checklist before requesting a review

Please delete options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: backend/models/databases/supabase/knowledge.py [hunk 1]
```diff
@@ -73,8 +73,25 @@ def get_knowledge_by_id(self, knowledge_id: UUID) -> Knowledge:
...
-            .filter("knowledge_id", "eq", str(knowledge_id))
+            .filter("id", "eq", str(knowledge_id))
             .execute()
         ).data
 
         return Knowledge(**knowledge[0])
+
+    def get_all_knowledge_in_brain(self, brain_id: UUID) -> List[Knowledge]:
+        """
+        Get a knowledge by its id
```
Comment (mamadoudicko) [inline backend/models/databases/supabase/knowledge.py:84]: 👀

File: backend/models/databases/supabase/knowledge.py [hunk 2]
```diff
@@ -73,8 +73,25 @@ def get_knowledge_by_id(self, knowledge_id: UUID) -> Knowledge:
...
+            brain_id (UUID): The id of the brain
+        """
+        all_knowledge = (
+            self.db.from_("knowledge")
+            .select("*")
+            .filter("brain_id", "eq", str(brain_id))
+            .execute()
+        ).data
+
+        print("ALL KNOWLEDGE", all_knowledge)
```
Comment (mamadoudicko) [inline backend/models/databases/supabase/knowledge.py:95]: 👀

File: frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/DeleteKnowledge.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}: {
+  knowledge: Knowledge;
+}): JSX.Element => {
+  const { isDeleting, onDeleteKnowledge } = useKnowledgeItem();
+
+  const { currentBrain } = useBrainContext();
+
+  const canDeleteFile = currentBrain?.role === "Owner";
+
+  console.log("isDeleting", isDeleting);
```
Comment (mamadoudicko) [inline frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/DeleteKnowledge.tsx:21]: 👀

File: frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/DownloadUploadedKnowledge.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      window.URL.revokeObjectURL(blobUrl);
+    } catch (error) {
+      console.error("Error downloading the file:", error);
+    }
+  };
+
+  return (
+    <a
+      onClick={() => void downloadFile()}
+      style={{ display: "flex", flexDirection: "column", alignItems: "center" }}
```
Comment (mamadoudicko) [inline frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/DownloadUploadedKnowledge.tsx:42]: 👀

File: frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/UploadedKnowledgeItem.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+"use client";
+
+import { UploadedKnowledge } from "@/lib/types/Knowledge";
+
+import { DeleteKnowledge } from "./DeleteKnowledge";
+import { DownloadUploadedKnowledge } from "./DownloadUploadedKnowledge";
+
+export const UploadedKnowledgeItem = ({
+  knowledge,
+}: {
```
Comment (mamadoudicko) [inline frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/UploadedKnowledgeItem.tsx:10]: Seems to have many things in common with `CrawledKnowledgeItem`... 

Maybe managing `isUploadedKnowledge` in another way ?

File: frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+}): JSX.Element => {
+  return isUploadedKnowledge(knowledge) ? (
+    <UploadedKnowledgeItem knowledge={knowledge} />
+  ) : (
+    <CrawledKnowledgeItem knowledge={knowledge} />
+  );
+};
+
+KnowledgeItem.displayName = "KnowledgeItem";
+export default KnowledgeItem;
```
Comment (mamadoudicko) [inline frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/index.tsx:21]: 👀

File: frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/useKnowledgeItem.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        }),
+      });
+    } catch (error) {
+      publish({
+        variant: "warning",
+        text: t("errorDeleting", { fileName: knowledge_name, ns: "explore" }),
+      });
+      console.error(`Error deleting ${knowledge_name ?? ""}`, error);
+    }
+    setIsDeleting(false);
```
Comment (mamadoudicko) [inline frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/KnowledgeItem/useKnowledgeItem.ts:52]: Seems ok but inside `finally` should be better

File: frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/hooks/useKnowledgeTab.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+type useKnowledgeTabProps = {
+  brainId: UUID;
+};
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useKnowledgeTab = ({ brainId }: useKnowledgeTabProps) => {
+  const [allKnowledge, setAllKnowledge] = useState<Knowledge[]>([]);
+
+  const { getAllKnowledge } = useKnowledgeApi();
+  const { data: brainKnowledges, isLoading: isPending } = useQuery({
+    queryKey: [getKnowledgeDataKey(brainId)],
```
Comment (mamadoudicko) [inline frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/hooks/useKnowledgeTab.ts:18]: Nice

File: frontend/app/brains-management/layout.tsx [hunk 9]
```diff
@@ -19,7 +20,7 @@ const Layout = ({ children }: LayoutProps): JSX.Element => {
   return (
     <div className="relative h-full w-full flex justify-stretch items-stretch">
       <BrainsList />
-      {children}
+      <KnowledgeProvider>{children}</KnowledgeProvider>
```
Comment (mamadoudicko) [inline frontend/app/brains-management/layout.tsx:23]: `KnowledgeProvider` seems to have be added 2 times

Comment (mamadoudicko) [reply frontend/app/brains-management/layout.tsx:23]: `KnowledgeTab/index.tsx` too

File: frontend/lib/api/knowledge/config.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import { UUID } from "crypto";
+
+const brainDataKey = "quivr-knowledge";
```
Comment (mamadoudicko) [inline frontend/lib/api/knowledge/config.ts:3]: 👀

File: frontend/lib/context/KnowledgeProvider/knowledge-provider.tsx [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+export const KnowledgeContext = createContext<KnowledgeContextType | undefined>(
+  undefined
+);
+
+export const KnowledgeProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}): JSX.Element => {
+  const [allKnowledge, setAllKnowledge] = useState<Knowledge[]>([]);
```
Comment (mamadoudicko) [inline frontend/lib/context/KnowledgeProvider/knowledge-provider.tsx:21]: It appears that `allKnowledge` is not being used. A new state has been introduced in `useknowledgeTab`, which is conflicting with this

Right ?

**Review by mamadoudicko — APPROVED** [review]

File: frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/hooks/useKnowledgeTab.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+import { getKnowledgeDataKey } from "@/lib/api/knowledge/config";
+import { useKnowledgeApi } from "@/lib/api/knowledge/useKnowledgeApi";
+import { Knowledge } from "@/lib/types/Knowledge";
+
+type useKnowledgeTabProps = {
+  brainId: UUID;
+};
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useKnowledgeTab = ({ brainId }: useKnowledgeTabProps) => {
+  const [allKnowledge, setAllKnowledge] = useState<Knowledge[]>([]);
```
Comment (gozineb) [inline frontend/app/brains-management/[brainId]/components/BrainManagementTabs/components/KnowledgeTab/hooks/useKnowledgeTab.ts:14]: replace get and set from the context

**Pull request merged**